deepspeed/moe/utils.py

```diff
@@ -150,3 +150,29 @@ def split_params_into_different_moe_groups_for_optimizer(
 
 def is_moe_param_group(param_group):
     return param_group.get('moe', False)
+
+
+def configure_moe_param_groups(model_parameters: List):
```

Comment:
Is `model_parameters` always a list? Can it be other Iterable? I wonder if [0] always works.

Reply:
Good question, I was also curious if we could assume this. But if you trace back where model_parameters is coming from you can see we force it to be a list here: 

https://github.com/microsoft/DeepSpeed/blob/731fd682996a0af56879c033ae3baec3db844e71/deepspeed/runtime/engine.py#L304

Reply:
added an assert here just in case the function is used outside of engine for some reason later